src/promptflow/promptflow/_core/tools_manager.py

```diff
@@ -53,6 +53,22 @@ def collect_tools_from_directory(base_dir) -> dict:
     return tools
 
 
+def set_default_input_index_for_tool(tool):
+    if _should_add_index_for_tool(tool):
```

Comment:
Delete _should_add_index_for_tool method. Here we can directly use the _should_add_index method by import it from tool_meta_generator